io/eolearn/tests/test_geometry_io.py

```diff
@@ -0,0 +1,98 @@
+"""
+Credits:
+Copyright (c) 2021-> Matej Aleksandrov, Matej Batič (Sinergise)
+
+This source code is licensed under the MIT license found in the LICENSE
+file in the root directory of this source tree.
+"""
+import pytest
+import logging
+import os
+
+from eolearn.core import FeatureType
+from eolearn.io import GeoDBVectorImportTask, GeopediaVectorImportTask, VectorImportTask
+
+from sentinelhub import BBox, CRS
+from xcube_geodb.core.geodb import GeoDBClient
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+@pytest.mark.parametrize(
+    argnames='reproject, clip, n_features, bbox',
+    ids=['simple', 'bbox', 'bbox_full', 'bbox_smaller'],
+    argvalues=[
+        (False, False, 193, None),
+        (False, False, 193, BBox([857000, 6521500, 861000, 6525500], crs='epsg:2154')),
+        (True, True, 193,
+         BBox([657089.4234095092, 5071037.944679743, 661093.7825844937, 5075039.833417523], crs=CRS.UTM_31N)),
+        (True, True, 125,
+         BBox([657690.073230332, 5071637.340226217, 660493.1246978994, 5074440.391780451], crs=CRS.UTM_31N))
+    ])
+class TestVectorImportTask:
+
+    @pytest.mark.parametrize('path', [
+        os.path.join(os.path.dirname(os.path.realpath(__file__)), '../../../example_data/import-gpkg-test.gpkg'),
+        's3://eolearn-io/import-gpkg-test.gpkg'
```

Comment:
I don't have a permission for this bucket, neither does CI. Perhaps we can instead mock this with `moto`? This way we could be used our locally saved `import-gpkg-test.gpkg` in the background.

Reply:
My tries were unsuccessful. I didn't want to spend a lot of time on this, so I rather made it so that the test is skipped if the user cannot access the bucket.

Reply:
I'll check if I can find a solution for this. Because if we have tests for which only you have credentials to run then we'll forget about that and the tests will be usually skipped.

Reply:
I investigated this and it turns out:

- `fiona` uses `boto3` only to parse session information and not for interaction with S3 buckets. Instead it uses GDAL implementation of S3 client. Because of that `moto` cannot mock it.

- I checked how AWS tests are implemented in `fiona` ([here](https://github.com/Toblerity/Fiona/blob/master/tests/test_vfs.py#L185)). It turns out they also have a real bucket and credentials and they skip tests if credentials are missing.

Hence it is safe to say there is no easy way how to mock this and we can keep tests as they are now.